src/hotspot/share/opto/type.hpp

```diff
@@ -1799,14 +1879,32 @@ inline const TypeMetadataPtr *Type::isa_metadataptr() const {
 }
 
 inline const TypeKlassPtr *Type::isa_klassptr() const {
-  return (_base == KlassPtr) ? (TypeKlassPtr*)this : NULL;
+  return (_base >= KlassPtr && _base <= AryKlassPtr ) ? (TypeKlassPtr*)this : NULL;
 }
 
 inline const TypeKlassPtr *Type::is_klassptr() const {
-  assert( _base == KlassPtr, "Not a klass pointer" );
+  assert( _base >= KlassPtr && _base <= AryKlassPtr, "Not a klass pointer" );
   return (TypeKlassPtr*)this;
 }
 
+inline const TypeInstKlassPtr *Type::isa_instklassptr() const {
+  return (_base == InstKlassPtr) ? (TypeInstKlassPtr*)this : NULL;
+}
+
+inline const TypeInstKlassPtr *Type::is_instklassptr() const {
+  assert( _base == InstKlassPtr, "Not a klass pointer" );
```

Comment:
The extra whitespaces should be removed.

Reply:
done.